cranelift/assembler-x64/meta/src/generate/inst.rs

```diff
@@ -221,13 +222,18 @@ impl dsl::Inst {
                             let to_string = location.generate_to_string(op.extension);
                             fmtln!(f, "let {location} = {to_string};");
                         }
-                        // Fix up the mnemonic for locked instructions: we want to print
-                        // "lock <inst>", not "lock_<inst>".
-                        let inst_name = if self.mnemonic.starts_with("lock_") {
-                            &format!("lock {}", &self.mnemonic[5..])
+                        let inst_name = if self.custom.contains(Display) {
+                            if self.name().starts_with("lock") {
+                                display::lock(&self.mnemonic)
+                            } else {
+                                unimplemented!(
+                                    "Only lock instructions custom processing implemented for now"
+                                );
```

Comment:
This should look more like the logic with `visit` above: if we have a custom `display` function, then we emit something like:

```rust
fmtln!(f, "let inst_name = crate::custom::display::{}(self)", self.name());
```

And then, in this case, we would use `inst_name` later in the generated code instead of the already-known `inst_name` we have here.